File: c/tree/flatten-binary-tree-to-linked-list.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */

void flatten(struct TreeNode* root) {
    if(root == NULL)
        return ;

    struct TreeNode *left = root->left;
    struct TreeNode *right = root->right;

    root->left = NULL;
    root->right = left;

    flatten(left);
    flatten(right);

    while(root->right)
        root = root->right;

    root->right = right;
}

File: python/tree/flatten-binary-tree-to-linked-list.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def flatten(self, root):
        if not root:
            return

        left = root.left
        right = root.right

        root.left = None
        root.right = left

        self.flatten(left)
        self.flatten(right)

        while root.right:
            root = root.right

        root.right = right
